src/Data/List.purs

```diff
@@ -519,6 +521,13 @@ take = go Nil
   go acc _ Nil = reverse acc
   go acc n (x : xs) = go (x : acc) (n - 1) xs
 
+-- | Take the specified number of elements from the end of a list.
+-- |
+-- | Running time: `O(2n - m)` where `n` is the number of elements in list
+-- | and `m` is number of elements to take.
+takeEnd :: forall a. Int -> List a -> List a
+takeEnd n xs = drop (length xs - n) xs
```

Comment:
This involves two traversals of the list, which we should try to avoid. Let's try to write these using `foldr` instead please.

Reply:
Is it even possible for this function to have better complexity? 

in order to take `n` elements from the end we need to start iterating the list from the start till we reach `length - n` position and return the tail. So we require length (whole list iteration) and we need to determine the index we stop on  (`length - n` elements iteration)

Reply:
Yeah, ignore me, this is fine as it is. I thought we could do it in one pass with a buffer of size `n`, but it's probably not worth it. We can revisit later if necessary.

Reply:
Alternative implementation will be to first reverse list and then take. it's performance will be n + m. it will be better then current version when m is smaller then n/2.

Reply:
first reverse list and then take _and then reverse_

Reply:
oh yeh :D so that will be better when m is less then n/4. so never mind